### PR TITLE
Support windows clang and fixes

### PR DESCRIPTION
**Pull request by NIKEA-SOFT**

### CODE REVIEW

File: server/server.cpp [hunk 1]
```diff
@@ -31,9 +31,12 @@
 // Stuff to wait for input on stdin.
 #ifdef Q_OS_WIN
 #include <QMutexLocker>
+#include <io.h>
 #else
 #include <QtCore/QSocketNotifier>
+#include <unistd.h>
```
Comment (lmoureaux) [inline server/server.cpp:37]: Why do we need `unistd` on Linux and Mac?

Comment (NIKEA-SOFT) [reply server/server.cpp:37]: > Why do we need `unistd` on Linux and Mac?

https://github.com/longturn/freeciv21/blob/f779ae6b04739d613b5077c47aebc550f9657fc1/server/server.cpp#L276

Comment (lmoureaux) [reply server/server.cpp:37]: Thanks :) It's obviously already included transitively but better do it explicitly indeed.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -120,7 +120,48 @@ jobs:
...
             build/Windows-x86_64/Freeciv21-*-Windows-x86_64.exe
             build/Windows-x86_64/Freeciv21-*-Windows-x86_64.exe.sha256
-
+  windows_clang_msvc:
+    name: "Windows Clang"
+    runs-on: windows-latest
+    needs: clang-format
+    strategy:
+      matrix:
+        architecture: [Win32, x64]
```
Comment (jwrober) [inline .github/workflows/build.yaml:129]: Do we need both architectures here? Why not just x86?

File: ai/default/aitools.cpp [hunk 3]
```diff
@@ -743,7 +743,6 @@ void dai_unit_new_task(struct ai_type *ait, struct unit *punit,
       if (unit_owner(missile) == unit_owner(punit)
           && def_ai_unit_data(missile, ait)->task != AIUNIT_ESCORT
           && !unit_transported(missile)
-          && unit_owner(missile) == unit_owner(punit)
```
Comment (jwrober) [inline ai/default/aitools.cpp:746]: This does not seem to be related to the build tools update

Comment (lmoureaux) [reply ai/default/aitools.cpp:746]: It's the same condition as in the first line
